### PR TITLE
feature: PHP8.2 - Transformers - Support Disjunctive Normal Form Types

### PR DESCRIPTION
**Pull request by SpacePossum**
This fixes the DNF types support on the tokens level. It adds new CT types for the braces and fixes the other transformers that didn't handle the new PHP8.2 syntax properly.

### CODE REVIEW

File: src/Tokenizer/CT.php [hunk 1]
```diff
@@ -54,6 +54,8 @@ final class CT
     public const T_NAMED_ARGUMENT_COLON = 10033;
     public const T_FIRST_CLASS_CALLABLE = 10034;
     public const T_TYPE_INTERSECTION = 10035;
+    public const T_DISJUNCTIVE_NORMAL_FORM_TYPE_BRACE_OPEN = 10036;
+    public const T_DISJUNCTIVE_NORMAL_FORM_TYPE_BRACE_CLOSE = 10037;
```
Comment (jrmajor) [inline src/Tokenizer/CT.php:58]:
```suggestion
    public const T_TYPE_PARENTHESIS_OPEN = 10036;
    public const T_TYPE_PARENTHESIS_CLOSE = 10037;
```

See #5915.

File: tests/Tokenizer/Transformer/DisjunctiveNormalFormTypeBracesTransformerTest.php [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    function G():(A&C)|I;
+}
+class G {
+    public function X():(Z&Y)|P{}
+}
+',
+        ];
+    }
+
+    public function provideProcessAttributeCases(): iterable
```
Comment (jrmajor) [inline tests/Tokenizer/Transformer/DisjunctiveNormalFormTypeBracesTransformerTest.php:156]:
```suggestion
    public function provideProcessPropertyCases(): iterable
```

File: src/Tokenizer/CT.php (hunk 1 above)
Comment (SpacePossum) [reply src/Tokenizer/CT.php:58]: fine by me to call these `PARENTHESIS` , I would leave the DNF part in the name as these should not be confused by any other parenthesis

Comment (jrmajor) [reply src/Tokenizer/CT.php]: > I would leave the DNF part in the name as these should not be confused by any other parenthesis

Isn't that why we have `TYPE` in name? I'm OK with leaving it like that though :)

Comment (SpacePossum) [reply src/Tokenizer/CT.php]: yeah for now we have only one kind typing in PHP which is using `(` `)`, but I wouldn't be surprised if we get more in the future ;)

**Pull request closed without merging**